### ISSUE TITLE
NPE inside aggregator

### ISSUE THREAD

**Issue by tpoterba** [hail-is/hail#3276]
to replicate:
```python
In [2]: mt = hl.import_vcf('src/test/resources/sample.vcf')
In [3]: mt.aggregate_entries(hl.agg.counter(hl.agg.explode(mt.PL)))
```
```
Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 2.0 failed 1 times, most recent failure: Lost task 0.0 in stage 2.0 (TID 2, localhost, executor driver): java.lang.NullPointerException
	at is.hail.codegen.generated.C1.apply(Unknown Source)
	at is.hail.codegen.generated.C1.apply(Unknown Source)
	at is.hail.expr.CM$$anonfun$runWithDelayedValues$1.apply(CM.scala:84)
	at is.hail.expr.CM$$anonfun$runWithDelayedValues$1.apply(CM.scala:82)
	at is.hail.expr.AST$$anonfun$runAggregator$1.apply(AST.scala:270)
	at is.hail.expr.AST$$anonfun$runAggregator$1.apply(AST.scala:268)
	at is.hail.methods.Aggregators$$anonfun$11.apply(Aggregators.scala:304)
	at is.hail.methods.Aggregators$$anonfun$11.apply(Aggregators.scala:300)
	at is.hail.variant.MatrixTable$$anonfun$82$$anonfun$apply$64$$anonfun$apply$65.apply(MatrixTable.scala:1743)
	at is.hail.variant.MatrixTable$$anonfun$82$$anonfun$apply$64$$anonfun$apply$65.apply(MatrixTable.scala:1741)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at is.hail.annotations.UnsafeIndexedSeq.foreach(UnsafeRow.scala:51)
	at is.hail.variant.MatrixTable$$anonfun$82$$anonfun$apply$64.apply(MatrixTable.scala:1741)
	at is.hail.variant.MatrixTable$$anonfun$82$$anonfun$apply$64.apply(MatrixTable.scala:1734)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at is.hail.variant.MatrixTable$$anonfun$82.apply(MatrixTable.scala:1734)
	at is.hail.variant.MatrixTable$$anonfun$82.apply(MatrixTable.scala:1728)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:797)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:797)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jigold** [comment]
I've narrowed down where the problem is. `agg.explode` is not treating missing values correctly. This is true if both PL is missing or the entry is missing (filtered out). `agg.explode` is equivalent to `flatMap` aggregator in Scala.

This will pass:
`agg.counter(agg.explode(hl.empty_array(hl.tint32)))`

This will fail:
`agg.counter(agg.explode(hl.null(tarray(tint32))))`

This will pass:
```
e = mt.entries().select('locus', 'alleles', 's', 'PL')
e = e.filter(hl.is_defined(e.PL))
e.aggregate(hl.agg.counter(hl.agg.explode(e.PL)))
```

I tried looking at the `flatMap` function registry function. It looks like the non-aggregator version has `flattenOrNull`, but not the aggregator version. @danking @catoverdrive can you look at the `flatMap` code both in the function registry and the IR to make sure missing values are handled correctly?

**Comment by catoverdrive** [comment]
I think I found the place this is wonky, but I've never touched the `CM` stuff so I'm not super sure how to fix it.

**Comment by jigold** [comment]
Fixed by #3348.
